chain/client/src/client.rs

```diff
@@ -1643,6 +1643,8 @@ impl Client {
 
         self.shards_manager_adapter
             .send(ShardsManagerRequestFromClient::CheckIncompleteChunks(*block.hash()));
+
+        self.process_ready_orphan_chunk_state_witnesses(&block);
```

Comment:
> Sanity check - the pool is not synchronized but the calls to the pool are because all of them are done from the client thread - am I about right?

All functions which mutate the orphan pool have `&mut Client`, so there's no need for any synchronization, Rust makes sure that there are no race conditions.

> Does the order of invocations of process_ready_orphan_chunk_state_witnesses and produce_chunks matter? I may be off by one, forgive me if so :)

I think it doesn't matter. If we have a witness which is waiting for block at height `H`, then this witness contains a chunk header with height `H + 1`, which means that the the chunk at height `H + 1` has already been produced. It shouldn't interfere with this node producing another chunk at height `H + 1`.